Adapter/GADMAdapterMoPub.m

```diff
@@ -1,263 +0,0 @@
-    self.bannerAd.delegate = nil;
-    self.interstitialAd.delegate = nil;
-}
-
-- (void)presentInterstitialFromRootViewController:(UIViewController *)rootViewController {
-    if (self.interstitialAd.ready) {
-        [self.interstitialAd showFromViewController:rootViewController];
-    }
-}
-
-#pragma mark MoPub Ad Interstitial delegate methods
-
-- (void)interstitialDidLoadAd:(MPInterstitialAdController *)interstitial {
-    [self.connector adapterDidReceiveInterstitial:self];
-}
-
-- (void)interstitialDidFailToLoadAd:(MPInterstitialAdController *)interstitial {
-    NSError *adapterError = [NSError errorWithDomain:adapterErrorDomain code:0 userInfo:nil];
-    [self.connector adapter:self didFailAd:adapterError];
-}
-
-- (void)interstitialWillAppear:(MPInterstitialAdController *)interstitial {
-    [self.connector adapterWillPresentInterstitial:self];
-}
-
-- (void)interstitialWillDisappear:(MPInterstitialAdController *)interstitial {
-    [self.connector adapterWillDismissInterstitial:self];
-}
-
-- (void)interstitialDidDisappear:(MPInterstitialAdController *)interstitial {
-    [self.connector adapterDidDismissInterstitial:self];
-}
-
-
-//Loading Banner Ads
-
-- (void)getBannerWithSize:(GADAdSize)adSize {
-
-  self.bannerAd = [[MPAdView alloc] initWithAdUnitId:[self.connector credentials][@"ad_unit"] size: CGSizeMake(adSize.size.width, adSize.size.height) ] ;
-
-  self.bannerAd.delegate = self;
-  self.bannerAd.adUnitId = self.connector.publisherId;
-
-  [self.bannerAd loadAd];
-  NSLog(@"Requesting banner from Sample Ad Network");
-}
-
-
-- (BOOL)isBannerAnimationOK:(GADMBannerAnimationType)animType {
-    return YES;
-}
-
-// Loading Native Ads
-
-- (void)getNativeAdWithAdTypes:(NSArray *)adTypes options:(NSArray *)options {
-  
-  MPStaticNativeAdRendererSettings *settings = [MPStaticNativeAdRendererSettings alloc];
-  
-  MPNativeAdRendererConfiguration *config = [MPStaticNativeAdRenderer rendererConfigurationWithRendererSettings:settings];
-  NSString *moPubPublisherId = [self.connector credentials][@"ad_unit"]; // for facebook testing @"1ceee46ba9744155aed48ee6277ecbd6"; //
-    
-  MPNativeAdRequest *adRequest = [MPNativeAdRequest requestWithAdUnitIdentifier: moPubPublisherId rendererConfigurations:@[config]];
-                                                
-  MPNativeAdRequestTargeting *targeting = [MPNativeAdRequestTargeting targeting];
-    CLLocation *currentlocation = [[CLLocation alloc] initWithLatitude:self.connector.userLatitude longitude:self.connector.userLongitude];
-    targeting.location = currentlocation;
-    
-  [adRequest startWithCompletionHandler:^(MPNativeAdRequest *request, MPNativeAd *response, NSError *error) {
-      if (error) {
-          [self.connector adapter:self didFailAd:error];
-      }
-      else {
-          self.nativeAd = response;
-          self.nativeAd.delegate = self;
-      
-          if(options!=nil){
-    
-              for (GADNativeAdImageAdLoaderOptions *imageOptions in options) {
```

Comment:
hm interesting, it's not necessarily a redundant check. just a weird check indicating not-ideal data models. can you show me their sample code?